core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java

```diff
@@ -80,14 +83,33 @@
 	 * @return
 	 */
 	default boolean isJobRunning(String jobKey) {
+		return isJobRunning(jobKey, (job) -> true); 
+	}
+	
+	/**
+	 * Returns <code>true</code> if any job present with the given jobKey in {@link RemoteJobEntry#isRunning()} state and matches the given predicate, <code>false</code> otherwise.
+	 * 
+	 * @param jobKey - the logical key assigned to the job
+	 * @param predicate - the predicate filter to further customize the search process
+	 * @return
+	 */
+	default boolean isJobRunning(String jobKey, Predicate<RemoteJobEntry> predicate) {
+		checkNotNull(predicate, "Predicate should not be null");
 		// check first if this context is running inside a job with the given jobKey
 		Optional<RemoteJob> job = optionalService(RemoteJob.class);
 		if (job.isPresent() && Objects.equals(jobKey, job.get().getKey())) {
 			return true;
```

Comment:
The predicate is not applied to the current job.

Reply:
Right, thanks! 😅